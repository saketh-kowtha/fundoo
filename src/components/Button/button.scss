@import "../../../index.scss";
//Scss for button
.btn{
    border: none;
    border-radius: 4px;
    color: $buttonFgColor;
    font-size: $buttonFontSize;
    font-weight: bold;
    cursor: pointer;
    outline: none;
    padding: 12px 28px;
    background-color: $buttonBgColor;
}

.btn[type="small"]{
    font-size: calc($buttonFontSize - 6);
    padding: 8px 18px;

}