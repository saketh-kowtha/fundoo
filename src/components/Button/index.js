/**
 * @author Kowtha Saketh
 * @description Button Component
 */

import React from 'react'

import './button.scss'
/**
 * @name Button
 * @param {*} props 
 * @description Button component
 */
const Button = (props) => {
    return <button className="btn" {...props}>
        {props.children}
    </button>
}

export default Button

