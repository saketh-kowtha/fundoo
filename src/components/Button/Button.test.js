import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './index';

describe('Button', () => {
    const component = mount(<Button type="small">Button</Button>);
    
    it('should have class btn ', () => {
        let child = component.find("button").text()
        expect(child).toEqual("Button")        
    });

    it('Should have Children ', () => {
        let prop = component.find("button").prop("className")
        expect(prop).toBe("btn")
    })
   
    it("should render Correctly", () => {
        expect(component).toMatchSnapshot()
    })

    it("should update Correctly", () => {
        const component2 = mount(<Button type="small">Click</Button>);
        expect(component2).toMatchSnapshot()
    })

});