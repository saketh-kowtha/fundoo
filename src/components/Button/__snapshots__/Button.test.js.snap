// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Correctly 1`] = `ReactWrapper {}`;

exports[`Button should render correctly  1`] = `ShallowWrapper {}`;

exports[`Button should update Correctly 1`] = `ReactWrapper {}`;
